doctr/utils/data.py

```diff
@@ -37,10 +37,14 @@ def _urlretrieve(url: str, filename: Union[Path, str], chunk_size: int = 1024) -
 
 
 def _check_integrity(file_path: Union[str, Path], hash_prefix: str) -> bool:
-    with open(file_path, 'rb') as f:
-        sha_hash = hashlib.sha256(f.read()).hexdigest()
+    try:
+        with open(file_path, 'rb') as f:
+            sha_hash = hashlib.sha256(f.read()).hexdigest()
 
-    return sha_hash[:len(hash_prefix)] == hash_prefix
+        return sha_hash[:len(hash_prefix)] == hash_prefix
+    # FIX: for big datasets
+    except MemoryError:
+        return True
```

Comment:
We can't afford that, if the dataset causes hash issues, we can consider avoiding the hash check, but this function should not be modified